[db] Prepared statements for queue items + some refactoring

ejurgensen opened the pull request:
@chme in this PR I have made some changes so that the queue functions in db.c are more similar to the functions for files and playlists. As part of that, adding and updating queue items now use prepared statements. I also tried to consolidate the mapping of fields between mfi/dbmfi and queue items into a single table.

While testing this I came across an issue that I would like your view on. It relates to the query_limit of playlists, where I saw that in my database there are no less than three different default values used: -1, 0 and UINT32_MAX. The -1 comes from your PR #525 (specifically commit 6ceede44f49cb0c9500069b1844c2322205ea529), and the UINT32_MAX is a result of my later implementation of prepared statements, where bind_generic() casts pli->query_limit to uint32_t when saving to the database. Finally I guess the 0 just comes from a calloc'ed pli getting saved.

Part of the issue is that db.c only has the data type DB_TYPE_INT for 32 bit, and that is handled as uint32, because query_limit is the only 32 bit int field that isn't unsigned.

Three defaults obviously isn't so good, and it is also dangerous that pli->query_limit is the only int32. So one alternative is to go back to unsigned with 0 as default. I couldn't quite figure out if that would work with the smart playlists - can you see that? If not, I think query_limit should be changed to int64, then we would be down to two defaults, -1 and 0, and query_limit would no longer have an "unusual" data type.

Of course, should you notice any other issues with these changes then please let me know!

chme commented:
A limit of "-1" is what sqlite accepts to return all rows. Having to defaults does not sound good. I guess introducing a new type for a signed 32 bit integer is not worth the effort for this one column. I would then prefer to change the default to 0 (meaning no limit). But in this case we should make sure a limit == 0 won't add a limit clause (or a limit clause with -1) in `db_build_query_clause` to the query.

The changes in this MR look good to me.

ejurgensen commented:
> we should make sure a limit == 0 won't add a limit clause (or a limit clause with -1) in db_build_query_clause to the query

Do you mean like this (in db_build_query_clause)?

```
      case I_FIRST:
	if (qp->limit)
	  qc->index = sqlite3_mprintf("LIMIT %d", qp->limit);
	else
	  qc->index = sqlite3_mprintf("");
	break;
```
To work with pli->query_limit == 0 then I suppose this one from  db_build_query_plitems_smart() also needs modification, but I think that might be all.
```
count = sqlite3_mprintf("SELECT COUNT(*) FROM files f %s AND %s LIMIT %d;", qc->where, pli->query, pli->query_limit);
```

chme commented:
Yes, that should be the two places to change.
I would change the if statement to 

```
if (qp->limit > 0)
```

imo, that makes it a bit more obvious that 0 and negative values do not limit the result sets.

ejurgensen commented:
I have now changed the default query_limit to 0, and also made it a uint32. I notice the smart pl scanner has -1 as default limit for smartpl.limit, but I didn't change that. Instead I just changed the mapping in filescanner_smartpl.c, so that only positive values of smartpl.limit are copied to pli.query_limit.

I plan to merge it after the weekend, unless you notice any issues.
